src/cr/cube/cube.py

```diff
@@ -712,13 +713,12 @@ def raw_cube_array(self):
     def requires_array_transposition(self):
         """True if raw cube array needs transposition, False otherwise.
 
-        When one of the dimension type is a NUM_ARRAY, it's cube_array values have to be
+        When one of the dimension type is a NUM_ARRAY, its cube_array values have to be
         transposed. This business rule needs to drive correctly tabbook and deck exports
         when a numeric array dimension is expressed.
         """
-        return len(self._all_dimensions) >= 2 and any(
-            [d.dimension_type == DT.NUM_ARRAY for d in self._all_dimensions]
-        )
+        dimension_types = [d.dimension_type for d in self._all_dimensions]
```

Comment:
I don't see why this is necessary. I _think_ that `NUM_ARRAY` should appear in `apparent_dimensions`, and thus its type in `self.dimension_types`. Is this not so?

Reply:
it is true, but we don't have dimension_types property accessible there 🤷🏽‍♂️